docs/UIO.md

```diff
@@ -0,0 +1,172 @@
+and label (txt) are stored in a single compressed package in sequence.
+
+### Chain IO
+Inspired by TFRecord chain IO, UIO also adopts chain implementation. In practical use, chain IO is more flexible,
+easier to expand and easier to debug. TFRecord IO example as follows,
+```python
+def read_dataset(filename, batch_size):
+    dataset = tf.data.TFRecordDataset(filename)
+    dataset = dataset.map(_parse_image_function, num_parallel_calls=tf.data.experimental.AUTOTUNE)
+    dataset = dataset.shuffle(500)
+    dataset = dataset.batch(batch_size, drop_remainder=True)
+    dataset = dataset.repeat()
+    dataset = dataset.prefetch(buffer_size=tf.data.experimental.AUTOTUNE)
+    return dataset
+```
+Refer to TFRecord IO, the UIO dataflow in WeNet is designed as the figure below:
+
+![UIO dataflow](./images/UIO_dataflow.png)
+
+It includes the following modules:
+- tokenize module: convert the label into specify modeling unit(egs: char or BPE).
+
+- filter module: filter out unqualified training data.
+
+- resample module: optional resampling of training data.
+
+- compute_fbank module: fbank feature extraction.
+
+- spec_augmentation module: feature enhancement.
+
+- shuffle module: disrupt local data.
+
+- sort module: sort local data.
+
+- batch module: organize multiple pieces of data into batch.
+
+- padding module: padding data in the same batch.
+
+what's more, There are several parameters to note. first, **shuffle buffer** and **sort buffer** in buffer size:
+* Shuffle buffer: shuffle data. It is recommended that the size of this buffer be larger than
+the number of data contained in a single shard. Each time it is equivalent to shuffling data between two shards,
+which increases the randomness of the data.(egs: if each shard contains 1000, you can set shuffle buffer as 1500)
+* Sort buffer: sort the data according to the number of frames. This operation is very important and can greatly
+improve the training speed.
+
+then, **Prefetch**:
+Prefetch is used in the Pytorch Dataloader to pre-read data. The granularity of prefetch is the batch of final training.
+The default parameter is 2, that is, the data of two batches will be pre-read by default. In the design of the UIO,
+due to the existence of the pre buffer, the pre-read data may already be in the buffer, so there is no real pre read.
+Only when the data in the buffer is insufficient during the next training can the buffer be filled on the fly.
+At this time, the training is blocked on the read data. In a word, when the prefetch is very small, the training will
+block reading data in part of the time, because the previous data is still in cache. Therefore, you can set a large
+prefetch to avoid this problem.
+
+
+## Validation experiments
+At present, we have verified the accuracy of UIO on aishell (200 hours) and wenetspeech (10000 hours) data respectively.
+### Aishell(raw vs shard)
+
+![UIO Aishell](./images/UIO_aishell_cer.png)
+
```

Comment:
Markdown table is prefered when comparing numbers，could you replace this image with a table ?

Reply:
> 

ok. I replace UIO_aishell_cer.pmg by markdown table, and retain UIO_wenetspeech_cer.png in order to keep the reference trace.